src/new/sendTransaction.js

```diff
@@ -152,8 +163,45 @@ class SendTransaction extends EventEmitter {
    * Start object (submit job)
    */
   start() {
+    this.updateOutputSelected(true);
     this.submitJob();
   }
+
+  /**
+   * Update the outputs of the tx data in localStorage to set 'selected_as_input'
+   * This will prevent the input selection algorithm to select the same input before the
+   * tx arrives from the websocket and set the 'spent_by' key
+   *
+   * @param {boolean} selected If should set the selected parameter as true or false
+   * @param {Object} store Optional store to use to select the localStorage
+   *
+   **/
+  updateOutputSelected(selected, store = null) {
+    if (store !== null) {
+      storage.setStore(store);
+    }
+
+    const walletData = wallet.getWalletData();
+    const historyTransactions = 'historyTransactions' in walletData ? walletData.historyTransactions : {};
+    const allTokens = 'allTokens' in walletData ? walletData.allTokens : [];
+
+    // Before sending the tx to be mined, we iterate through the inputs and set selected_as_input
+    for (const input of this.data.inputs) {
+      historyTransactions[input.tx_id].outputs[input.index]['selected_as_input'] = selected;
+    }
+    wallet.saveAddressHistory(historyTransactions, allTokens);
+
+    if (selected) {
+      // Schedule to set all those outputs as not selected later
+      const myStore = storage.store;
+      this._unmark_as_selected_timer = setTimeout(() => {
```

Comment:
You still have to check whether it is null or not. Otherwise you can end up setting multiple timers.

Reply:
Done